ddl/db_change_test.go

```diff
@@ -1991,3 +1992,120 @@ func (s *stateChangeSuite) TestRestrainDropColumnWithIndex() {
 	tk.MustExec("alter table t drop column a;")
 	tk.MustExec("drop table if exists t;")
 }
+
+func TestParallelRenameTable(t *testing.T) {
+	store, d, clean := testkit.CreateMockStoreAndDomain(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("create database test2")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int default 0, b int default 0, key idx((b+1)))")
+	tk1 := testkit.NewTestKit(t, store)
+	tk1.MustExec("use test")
+	tk3 := testkit.NewTestKit(t, store)
+	tk3.MustExec("use test")
+
+	var concurrentDDLQueryPre string
+	var concurrentDDLQuery string
+	firstDDL := true
+
+	var wg sync.WaitGroup
+	var checkErr error
+	d2 := d.DDL()
+	originalCallback := d2.GetHook()
+	defer d2.SetHook(originalCallback)
+	callback := &ddl.TestDDLCallback{Do: d}
+	callback.OnJobRunBeforeExported = func(job *model.Job) {
```

Comment:
Could we use `testControlParallelExecSQL` directly?

Reply:
It needs to make some effort to use testControlParallelExecSQL.
1. testControlParallelExecSQL relies on stateChangeSuite.
2. It doesn't fit the new concurrent DDL framework.
3. It can't promise that ddl2 is run after dd1.